source/patterns/@aws-solutions-constructs/aws-alb-fargate/lib/index.ts

```diff
@@ -0,0 +1,247 @@
+   * why loadBalancerProps and privateHostedZoneProps can't include a VPC). Providing
+   * both this and existingVpc is an error.
+   *
+   * @default - none
+   */
+  readonly vpcProps?: ec2.VpcProps;
+  /**
+   * An existing VPC in which to deploy the construct. Providing both this and
+   * vpcProps is an error. If the client provides an existing load balancer and/or
+   * existing Private Hosted Zone, those constructs must exist in this VPC.
+   *
+   * @default - none
+   */
+  readonly existingVpc?: ec2.IVpc;
+  /**
+   * Whether to turn on Access Logs for the Application Load Balancer. Uses an S3 bucket
+   * with associated storage costs. Enabling Access Logging is a best practice.
+   *
+   * @default - true
+   */
+  readonly logAlbAccessLogs?: boolean;
+  /**
+   * Optional properties to customize the bucket used to store the ALB Access
+   * Logs. Supplying this and setting logAccessLogs to false is an error.
+   *
+   * @default - none
+   */
+  readonly albLoggingBucketProps?: s3.BucketProps;
+  /**
+   * Whether the construct is deploying a private or public API. This has implications for the VPC and ALB.
+   *
+   * @default - none
+   */
+  readonly publicApi: boolean;
+  /**
+   * Optional properties to create a new ECS cluster
+   */
+  readonly clusterProps?: ecs.ClusterProps;
+  /**
+   * The arn of an ECR Repository containing the image to use
+   * to generate the containers
+   *
+   * format:
+   *   arn:aws:ecr:[region]:[account number]:repository/[Repository Name]
+   */
+  readonly ecrRepositoryArn?: string;
+  /**
+   * The version of the image to use from the repository
+   *
+   * @default - none
+   */
+  readonly ecrImageVersion?: string;
+  /**
+   * Optional props to define the container created for the Fargate Service
+   *
+   * @default - see fargate-defaults.t
+   */
+  readonly containerDefinitionProps?: ecs.ContainerDefinitionProps | any;
+  /**
+   * Optional props to define the Fargate Task Definition for this construct
+   *
+   * @default - see fargate-defaults.ts
+   */
+  readonly fargateTaskDefinitionProps?: ecs.FargateTaskDefinitionProps | any;
+  /**
+   * Optional properties to override default values for the Fargate service.
+   * Service will set up in the Public or Isolated subnets of the VPC by default,
+   * override that (e.g. - choose Private subnets) by setting vpcSubnets on this
+   * object.
+   *
+   * @default - see core/lib/fargate-defaults.ts
+   */
+  readonly fargateServiceProps?: ecs.FargateServiceProps | any;
+  /**
+   * A Fargate Service already instantiated (probably by another Solutions Construct). If
+   * this is specified, then no props defining a new service can be provided, including:
+   * ecrImageVersion, containerDefintionProps, fargateTaskDefinitionProps,
+   * ecrRepositoryArn, fargateServiceProps, clusterProps, existingClusterInterface
+   *
+   * @default - none
+   */
+  readonly existingFargateServiceObject?: ecs.FargateService;
```

Comment:
Right now it could probably be the interface - but we're just scratching the service of the container ecosystem and can't see very far into the future. 6-12 months from now we can go back and change this to IFargateService without it being a breaking change. If we go with the interface now and find out in the future we need to use the object we would have to make a breaking change.